File: leetcode-submissions/1046-max-consecutive-ones-iii/solution.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        //subarray, sliding windonw, nums, k, flip k 0's
        //record the number of zeros
        int left = 0;
        int currZeroes = 0;
        int ans = 0; //size of the subarray
        for (int right = 0; right < nums.length; right++) {
            if (nums[right] == 0) {
                currZeroes++;
            }
            
            while(currZeroes > k) {
                if (nums[left] == 0) {
                    currZeroes--;
                }
                left++;
            }
            
            ans = Math.max(ans, right - left + 1);
        }
        return ans;
    }
}

File: leetcode-submissions/1046-max-consecutive-ones-iii/solution.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        left = right = 0
        
        current_number_of_zeroes = 0
        ans = 0        
        
        while right < len(nums):
            if nums[right] == 0:
                current_number_of_zeroes += 1
                
            if current_number_of_zeroes > k:
                while nums[left] != 0:
                    left += 1
                left += 1
                current_number_of_zeroes -= 1
                
            current_length = (right - left) + 1
            ans = max(ans, current_length)
            right += 1
            
        return ans
        
